src/react-hook-form/fields/ControlOAuth.tsx

```diff
@@ -0,0 +1,77 @@
+import LoadingButton from '@mui/lab/LoadingButton';
+import FormHelperText from '@mui/material/FormHelperText';
+import Stack from '@mui/material/Stack';
+import { useContext, useEffect } from 'react';
+import Box from '@mui/system/Box';
+import { useOauthConnector } from '../../hooks';
+import { cloudQueryApiBaseUrl } from '../../utils';
+import { getOauthSuccessBaseUrl } from '../../utils/getOauthSuccessBaseUrl';
+import { useFormContext } from 'react-hook-form';
+import { PluginContext } from '../../context/plugin';
+
+/**
+ * This component is a renders an OAuth authentication button and handles the data transfer process.
+ *
+ * @public
+ */
+export function ControlOAuth({
+  pluginUiMessageHandler,
+}: {
+  pluginUiMessageHandler: any; // TODO: delete after iframe deprecation
+}) {
+  const form = useFormContext();
+  const { plugin, teamName, config } = useContext(PluginContext);
```

Comment:
That could a hook like:
```suggestion
  const { plugin, teamName, config } = usePluginContext()
```